wikimedia/wikipedia-ios
# [YiR] Feature announcement

**Phabricator:** https://phabricator.wikimedia.org/T376316

### Notes
* This PR adds logic for presenting the YiR feature
* It timeboxes the presentation to a deadline, code to be removed after this date
* For presenting the iPad popover pointing to the profile button, we're creating a CGRect and not using the UIBArButtonItem. That was causing a crash, probably related to our custom nav bar

### Test Steps
1. Set your device to a valid region (Italy or France)
2. Fresh install the app and login
3. Set the language to Italian or French
4. Read articles and generate 6+ article views in the History tab.
5. Go to developer settings and toggle on Year in Review feature flag.
6. Run the app and background it on the explore feed
7. Open the app again. You should see the modal. Tapping the "Continue" button should take you to the feature
8. Fresh install the app again, repeat the previous steps
9. Go to an article
10. You should see the modal
11. Go back to another article - it should not appear again
12. Repeat all the steps on iPad
